test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/InstallCommandTests.cs

```diff
@@ -255,5 +257,51 @@ public static CommandRunnerResult RunInstall(string nugetExe, SimpleTestPathCont
 
             return r;
         }
+
+        [Fact]
+        public async Task Install_WithHttpSource_Warns()
+        {
+            // Arrange
+            using var pathContext = new SimpleTestPathContext();
+            // Set up solution, project, and packages
+            var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+            var packageA = new SimpleTestPackageContext("a", "1.0.0");
+            await SimpleTestPackageUtility.CreateFolderFeedV3Async(pathContext.PackageSource, packageA);
+            var packageAPath = Path.Combine(pathContext.PackageSource, "a", "1.0.0", "a.1.0.0.nupkg");
+
+            pathContext.Settings.AddSource("http-feed", "http://api.source/api/v2");
+            pathContext.Settings.AddSource("https-feed", "https://api.source/index.json");
+
+            var net461 = NuGetFramework.Parse("net461");
+
+            var projectA = SimpleTestProjectContext.CreateLegacyPackageReference(
+                "a",
+                pathContext.SolutionRoot,
+                net461);
+
+            Util.CreateFile(Path.GetDirectoryName(projectA.ProjectPath), "packages.config",
+@"<packages>
+  <package id=""A"" version=""1.0.0"" targetFramework=""net461"" />
+</packages>");
+
+            solution.Projects.Add(projectA);
+            solution.Create(pathContext.SolutionRoot);
+
+            // Act
+
+            var args = new string[]
+            {
+                Path.Combine(Path.GetDirectoryName(projectA.ProjectPath), "packages.config"),
+                "-OutputDirectory",
+                Path.Combine(pathContext.WorkingDirectory, "packages")
+            };
+
+            // Act
+            var result = RunInstall(_nugetExePath, pathContext, expectedExitCode: 0, additionalArgs: args);
+
+            // Assert
+            result.Success.Should().BeTrue();
```

Comment:
Maybe add the validaiton of `                Assert.Contains("Added package 'x.1.0.0' to folder", result.AllOutput);` in here as well?

In particular, you want to make sure a PC restore is running.

Reply:
Sure.